backend/api/Areas/Admin/Controllers/AdministrativeAreaController.cs

```diff
@@ -0,0 +1,187 @@
+            return new JsonResult(_mapper.Map<Model.AdministrativeAreaModel[]>(adminAreas));
+        }
+
+        /// <summary>
+        /// GET - Returns a administrative area for the specified 'id' from the datasource.
+        /// </summary>
+        /// <param name="id">The unique 'id' for the admin area to return.</param>
+        /// <returns>The administrative area requested.</returns>
+        [HttpGet("{id}")]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(Api.Models.PageModel<Model.AdministrativeAreaModel>), 200)]
+        [ProducesResponseType(typeof(Api.Models.ErrorResponseModel), 400)]
+        [SwaggerOperation(Tags = new[] {"admin-administrative-area"})]
+        public IActionResult GetAdministrativeArea(int id)
+        {
+            var adminArea = _pimsAdminService.AdministrativeArea.Get(id);
+            return new JsonResult(_mapper.Map<Model.AdministrativeAreaModel>(adminArea));
+        }
+
+        /// <summary>
+        /// GET - Returns a paged array of administrative areas from the datasource.
+        /// </summary>
+        /// <param name="filter"></param>
+        /// <returns>Paged object with an array of administrative areas.</returns>
+        [HttpPost("filter")]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(Api.Models.PageModel<Model.AdministrativeAreaModel>), 200)]
+        [ProducesResponseType(typeof(Api.Models.ErrorResponseModel), 400)]
+        [SwaggerOperation(Tags = new[] {"admin-administrative-area"})]
+        public IActionResult GetAdministrativeAreas(EModel.AdministrativeAreaFilter filter)
+        {
+            var page = _pimsAdminService.AdministrativeArea.Get(filter);
+            var result = _mapper.Map<Api.Models.PageModel<Model.AdministrativeAreaModel>>(page);
+            return new JsonResult(result);
+        }
+
+        /// <summary>
+        /// PUT - Update an administrative area in the datasource.
+        /// </summary>
+        /// <param name="model">The administrative area model.</param>
+        /// <returns>The administrative area updated.</returns>
+        [HttpPut("{id}")]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(Model.AdministrativeAreaModel), 200)]
+        [ProducesResponseType(typeof(Api.Models.ErrorResponseModel), 400)]
+        [SwaggerOperation(Tags = new[] {"admin-administrative-area"})]
+        public IActionResult UpdateAdministrativeAreaAsync([FromBody] Model.AdministrativeAreaModel model)
+        {
+            var entity = _mapper.Map<Entity.AdministrativeArea>(model);
+            try
+            {
+                _pimsAdminService.AdministrativeArea.Update(entity);
+
+                var adminArea = _mapper.Map<Model.AdministrativeAreaModel>(entity);
+                return new JsonResult(adminArea);
+            }
+            catch (DbUpdateException ex)
+            {
+                if (ex.InnerException?.Message.Contains("duplicate") ?? false)
```

Comment:
Probably a little broad only searching for a single word "duplicate", but for our use-case it'll be fine.

Reply:
The error says adding, but here you're updating.